### ISSUE TITLE
Az cli task fails with "[error]Argument list too long"

### ISSUE THREAD

**Issue by tokleine** [Azure/azure-cli#22294]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Related command**
I have whitelist the IP of my DevOps Host to be able to use it call my Function App.

**Describe the bug**
When trying to whitelist the host my azcli task in my DevOps pipeline fails intermittently, without any clear indication why.

This is the output I get:
```##[debug]Evaluating condition for step: 'Whitelist this host IP'
##[debug]Evaluating: SucceededNode()
##[debug]Evaluating SucceededNode:
##[debug]=> True
##[debug]Result: True
Starting: Whitelist this host IP
==============================================================================
Task         : Azure CLI
Description  : Run Azure CLI commands against an Azure subscription in a PowerShell Core/Shell script when running on Linux agent or PowerShell/PowerShell Core/Batch script when running on Windows agent.
Version      : 2.198.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/azure-cli
==============================================================================
##[debug]Using node path: /home/vsts/agents/2.202.1/externals/node10/bin/node
##[error]Argument list too long
##[debug]System.ComponentModel.Win32Exception (7): Argument list too long
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Microsoft.VisualStudio.Services.Agent.Util.ProcessInvoker.ExecuteAsync(String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, InputQueue`1 redirectStandardIn, Boolean inheritConsoleHandler, Boolean keepStandardInOpen, Boolean highPriorityProcess, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Agent.ProcessInvokerWrapper.ExecuteAsync(String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, InputQueue`1 redirectStandardIn, Boolean inheritConsoleHandler, Boolean keepStandardInOpen, Boolean highPriorityProcess, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Agent.Worker.Handlers.DefaultStepHost.ExecuteAsync(String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, Boolean inheritConsoleHandler, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Agent.Worker.Handlers.NodeHandler.RunAsync()
   at Microsoft.VisualStudio.Services.Agent.Worker.TaskRunner.RunAsync()
   at Microsoft.VisualStudio.Services.Agent.Worker.StepsRunner.RunStepAsync(IStep step, CancellationToken jobCancellationToken)
Finishing: Whitelist this host IP
```

**To Reproduce**
pool:
  vmImage: ubuntu-latest

jobs:
- job: jobName
  steps:
  - checkout: self
  - task: AzureCLI@2
      displayName: 'Whitelist this host IP'
      name: whitelistHost
      inputs:
        azureSubscription: mySubscription
        scriptType: pscore
        scriptLocation: inlineScript
        inlineScript: | 
          Write-Host 'Fetching my IP ...'
          $pages = "https://api.ipify.org/","https://ipinfo.io/ip","https://ifconfig.me/ip","https://ident.me"
          foreach ($page in $pages)
          {
              try {$myIP = (Invoke-WebRequest -uri "$page").Content
                  az functionapp config access-restriction add -g myRG -n myAFA `
                    --rule-name devops-agent --action Allow --ip-address $myIP --priority 50
                  break}
              catch {Write-Host "Error setting IP"}
              finally {Write-Host "Done setting IPs"}
          }

**Expected behavior**
This normally works and sets the first IP it gets. I have multiple pages to increase my chances that I get an IP.

Any advice on how to fix this is highly appreciated. Or also to help me figure out what I am doing wrong :-)

**Comment by yonzhan** [comment]
route to CXP team

**Comment by tokleine** [comment]
Update: I found the issue myself. 

In a previous step, I pulled the content of a .json file into an env variable. It turned out that this env variable was _just_ small enough to fit into the available memory pool, but the subsequent command didnt fit anymore. Thus I am now using a smaller .json to fill my env variable in the previous step.

The reference that helped me find it is this: https://stackoverflow.com/questions/28865473/setting-environment-variable-to-a-large-value-argument-list-too-long
